### PR TITLE
Port formbot to rules/yaml format

### PR DESCRIPTION
**Pull request by akelad**
**Proposed changes**:
- ...

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by akelad** [comment]
@wochinge @Ghostvv this is what my port of the formbot looks like at the moment, don't think it's worth merging yet since a few things are gonna change on functionality. And the validation stuff is still missing
Not sure if there's any thoughts on how the stories/rules were split up

File: examples/formbot/data/rules.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    - intent: thankyou
+    - action: utter_noworries
+
+  - rule: Bot challenge
+    steps:
+    - ...
+    - intent: bot_challenge
+    - action: utter_iamabot
+
+  - rule: Request restaurant form
```
Comment (wochinge) [inline examples/formbot/data/rules.yml:20]: ?
```suggestion
  - rule: Submit restaurant form
```

Comment (akelad) [reply examples/formbot/data/rules.yml:20]: but also start it 🤔

Comment (wochinge) [reply examples/formbot/data/rules.yml:20]: Oh true. You combined both. Neat!

**Comment by akelad** [comment]
@Ghostvv since your neg-rules branch was merged i think we can merge this as well now, would be great if you can review. All the other updates that get made based on my feedback i can add as a second PR

File: examples/formbot/data/rules.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+rules:
+  - rule: Greet user
+    steps:
+    - ...
```
[the remark this answers is not in the document]
Comment (akelad) [reply examples/formbot/data/rules.yml:4]: did you change that behaviour already?

[the remark this answers is not in the document]
Comment (akelad) [reply examples/formbot/data/rules.yml:4]: i thought that was coming when the `conditions:` are implemented

[the remark this answers is not in the document]
Comment (akelad) [reply examples/formbot/data/rules.yml:4]: oh wait no you mean only make it at the beginning of the conversation. I mean, not really imo :D

[the remark this answers is not in the document]
Comment (Ghostvv) [reply examples/formbot/data/rules.yml:4]: yes, but in this example to showcase that capability

File: examples/formbot/data/rules.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    - intent: request_restaurant
+    - action: restaurant_form
+    - form: restaurant_form
+    - form: null
+    - action: utter_submit
+    - action: utter_slots_values
+
+  - rule: Chitchat
+    steps:
+      - ...
```
Comment (Ghostvv) [inline examples/formbot/data/rules.yml:32]: in other rules indentation is different

File: examples/formbot/data/stories.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+      - intent: request_restaurant
+      - action: restaurant_form
+      - form: restaurant_form
+      - intent: stop
+      - action: utter_ask_continue
+      - intent: affirm
+      - action: restaurant_form
+      - form: null
+    - action: utter_submit
+    - action: utter_slots_values
```
Comment (Ghostvv) [inline examples/formbot/data/stories.yml:13]: indentation

File: examples/formbot/data/rules.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (akelad) [reply examples/formbot/data/rules.yml:4]: i think we might have it in other bots though?

File: examples/formbot/data/stories.yml (hunk 4 above)
Comment (akelad) [reply examples/formbot/data/stories.yml:13]: you did this vova!

Comment (Ghostvv) [reply examples/formbot/data/stories.yml]: because in the rules it was like that!

Comment (Ghostvv) [reply examples/formbot/data/stories.yml]: why rules have different indentation?

Comment (akelad) [reply examples/formbot/data/stories.yml]: because yaml doesn't care about this

Comment (Ghostvv) [reply examples/formbot/data/stories.yml]: 🙈

**Comment by Ghostvv** [comment]
@tabergma

**Comment by sibbsnb** [comment]
@akelad 

When i said "not sure", it took as  - outdoor_seating: True

```
Your input ->  hi                                                                                                                                                                                                      
Hello! I am restaurant search assistant! How can I help?
Your input ->  need to eat                                                                                                                                                                                             
what cuisine?
Your input ->  english                                                                                                                                                                                                 
how many people?
Your input ->  3                                                                                                                                                                                                       
do you want to seat outside?
Your input ->  not sure                                                                                                                                                                                                
do you want to continue?
Your input ->  yes                                                                                                                                                                                                     
please provide additional preferences
Your input ->  nothing                                                                                                                                                                                                 
please give your feedback on your experience so far
Your input ->  hmm                                                                                                                                                                                                     
All done!
I am going to run a restaurant search using the following parameters:
 - cuisine: hmm
 - num_people: 3
 - outdoor_seating: True
 - preferences: nothing
 - feedback: hmm
```

File: examples/formbot/data/rules.yml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+      - intent: bot_challenge
+      - action: utter_iamabot
+
+  - rule: Request restaurant form
+    steps:
+      - ...
+      - intent: request_restaurant
+      - action: restaurant_form
+      - form: restaurant_form
+      - form: null
```
Comment (chkoss) [inline examples/formbot/data/rules.yml:26]:
```suggestion
      - active_loop: restaurant_form
      - active_loop: null
```

File: examples/formbot/data/stories.yml [hunk 6]
```diff
@@ -0,0 +1,24 @@
+stories:
+  - story: stop form + continue
+    steps:
+      - intent: request_restaurant
+      - action: restaurant_form
+      - form: restaurant_form
```
Comment (chkoss) [inline examples/formbot/data/stories.yml:6]:
```suggestion
      - active_loop: restaurant_form
```

File: examples/formbot/data/stories.yml [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+      - intent: stop
+      - action: utter_ask_continue
+      - intent: affirm
+      - action: restaurant_form
+      - form: null
```
Comment (chkoss) [inline examples/formbot/data/stories.yml:11]:
```suggestion
      - active_loop: null
```

File: examples/formbot/data/stories.yml [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+      - action: utter_submit
+      - action: utter_slots_values
+
+  - story: stop form + stop
+    steps:
+      - intent: request_restaurant
+      - action: restaurant_form
+      - form: restaurant_form
```
Comment (chkoss) [inline examples/formbot/data/stories.yml:19]:
```suggestion
      - active_loop: restaurant_form
```

File: examples/formbot/data/stories.yml [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+      - intent: stop
+      - action: utter_ask_continue
+      - intent: deny
+      - action: action_deactivate_form
+      - form: null
```
Comment (chkoss) [inline examples/formbot/data/stories.yml:24]:
```suggestion
      - active_loop: null
```

**Comment by akelad** [comment]
@chkoss i assume your PR is gonna be merged today? Because then I would just merge this PR now, it's getting on my last nerve with the constant master merges 😂

**Comment by chkoss** [comment]
Sure, go ahead! I hope the other one gets merged today, it's `status:ready-to-merge`, too, just depends on how many master merges it needs 😆

**Pull request merged**